tensorboard/webapp/metrics/data_source/metrics_data_source.ts

```diff
@@ -37,9 +39,21 @@ import {
   TagToRunSampledInfo,
   TimeSeriesRequest,
   TimeSeriesResponse,
+  PersistableSettings,
 } from './types';
 
 const HTTP_PATH_PREFIX = 'data/plugin/timeseries';
+const LOCAL_STORAGE_KEY = '_tb_global_settings';
+
+/**
+ * `declare` so it does not get mangled or mangled differently when
+ * compiler changes.
+ */
+declare interface SerializableSettings {
```

Comment:
Not sure if this TS idea works out, but if the entire purpose of having these is to make TS check serializability, could we drop `SerializableSettings` and `serializeSettings` and just ensure `PersistableSettings` is serializable via `PersistableSettings extend Record<string, string|number|boolean>`?

Reply:
> just ensure PersistableSettings is serializable via PersistableSettings extend Record<string, string|number|boolean>

I prefer it this way; something that is to be persisted should have its own interface while the data structure that weaves through internals have another one. Sure, we can make it congruent but, imo, that's exactly in the realm of "let's dedup just because they happen to be the same".